pkg/autoscaler/autoscaler.go

```diff
@@ -170,16 +178,22 @@ func (agg *perPodAggregation) calculateAverage(now time.Time) float64 {
 	if agg.probeCount == 0 {
 		return 0.0
 	}
-	return agg.accumulatedConcurrency / float64(agg.probeCount) * agg.usageRatio(now)
+	return agg.accumulatedConcurrency / float64(agg.probeCount) * agg.podWeight(now)
 }
 
-// Calculates the weighted pod count
-func (agg *perPodAggregation) usageRatio(now time.Time) float64 {
-	if agg.lameduckTime == nil {
-		return float64(1.0)
+// Calculates the pod weight. Assuming the latest stat time is the point when
+// pod became out of service.
+func (agg *perPodAggregation) podWeight(now time.Time) float64 {
+	if agg.isActivator {
+		return 1.0 // Activator pod weight is always 1
 	}
-	outOfService := now.Sub(*agg.lameduckTime)
-	return float64(1.0) - (float64(outOfService) / float64(agg.window))
+
+	outOfService := now.Sub(*agg.latestStatTime)
+	// Less than minOutOfServiceTime means no out of service
+	if outOfService <= minOutOfServiceTime {
+		outOfService = 0
```

Comment:
save on FP computation and just return 1.0 here?

Reply:
Done.